### ISSUE TITLE
selecting and deleting multiple viruses in one go.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dave-theunsub/clamtk#36]
```
Please suggest a way to delete detected viruses when multiple viruses are 
found, selecting and deleting each one is very tedious when (sometimes) viruses 
are found in hundreds.

I am using Mint 17
```

Original issue reported on code.google.com by `rusto...@gmail.com` on 17 Jan 2015 at 4:15

**Comment by setherith** [comment]
How about check boxes against each item and an apply to selection button to quarantine or delete checked items?

**Comment by LarryRink** [comment]
Check boxes sounds great. I backup my entire system in the clouds and on a thumb drive. Even if you offered this feature and included a large lengthy warning in red you may help alot of us out big time.
Thanks in advance.

**Comment by dave-theunsub** [comment]
I'll have a look regarding implementation.  Thanks for offering suggestions!

Thanks,
Dave M

**Comment by dave-theunsub** [comment]
@TinaRussell , okay, I will look at adding something like that for 5.28.

However, see what I wrote earlier about Windows systems:
https://github.com/dave-theunsub/clamtk/issues/36#issuecomment-84165590

All you need is one item that the scanner thinks is bad or a PUA and bad stuff can happen. Be careful if you're using it on Windows.

**Comment by eqjjh** [comment]
I agree with the above, at least why can we not highlight multiple items and delete/quarantine en masse? Can take an incredibly long amount of time, as others have mentioned.

**Comment by Discusseded** [comment]
Hey Dave, thanks so much for a great tool. I just started using this and as a power user who is new to Linux I have to say that while yes, I probably am capable of breaking Linux unintentionally, I am a thinking agent that can own up to my own mistakes. I understand you want to eliminate the grief of loss, but in reality you are adding grief to the process for every single scan.

I was shocked to see that a Ctrl+A, Ctrl+Select, or Shift+Select paradigm wasn't used in this app. If that's too Windows-y, why not reinstate the * All buttons but add a dialogue box warning of the consequences and offering a way to back out and assess the threats responsibly? I presume you probably did have such a setup so I have to wonder why it was removed. Too much hate mail from dum dums who broke their OS?

Here I sit with about 50 lines that I have to go through one-by-one, click Delete, click Ok, rinse and repeat. I can't even use the up or down keys, so I have to move my mouse up to the next item, click, move down and click Delete, move back up and click Ok, move to the next item, and so on. That's 3(n) actions plus the hand-eye coordination. Please let's be reasonable about this. Empower your users, don't punish them. Ignore the dum dums, they don't deserve the attention.

**Comment by dave-theunsub** [comment]
Hi @Discusseded ,

Thank you for the great feedback.

Older versions did have something similar where you could delete a lot (or all, I think) at once.  The problem was that there were, at times, many false positives that could result in a broken/crummy state... so it was removed.  I know that's not always great for everyone... but as this is a Linux tool, we have to be careful in that this type of scanning is different than on a Windows one.  Most malware threats will be found in a person's home directory but people will still want to scan the entire system, and that was also where a lot of false positives came from.  Rootkits were getting installed in /tmp and /dev, but it's usually advised to not scan /dev (and /proc and others) for various reasons.  Also, ClamAV isn't as great for detecting rootkits - that's for rkhunter and chkrootkit, which do that.

Anyway, I'm rambling, but I do like the idea with the results though. Not sure that would make anyone change their mind but it does seem nice with a good flow.

I'll look into it.

Thanks,
Dave M
